### PR TITLE
CP-1156 Docs updates, w_transport 12.2

### LINKED ISSUE #82: Update description in pubspec.yaml

### LINKED ISSUE #86: Add section about Client to README

### PR DESCRIPTION
**Pull request by evanweible-wf**
**DOCS ONLY CHANGE**

- Update package description (fixes #82)
- Add section about `Client` to readme (fixes #86)
- Adds documentation badge to readme

## Code Review
@trentgrover-wf
@maxwellpeterson-wf 
@dustinlessard-wf
@jayudey-wf

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -438,6 +442,35 @@ StreamedRequest request = new StreamedRequest()
 > broadcast stream) to avoid losing data.
 
 
+### HTTP Client
+An HTTP client acts as a single point from which many requests can be
+constructed. All requests constructed from a client will inherit headers, the
+`withCredentials` flag, and the timeout threshold from the client.
```
Comment (dustinlessard-wf) [inline README.md]: The trailing 'from a client' might be redundant.

**Comment by dustinlessard-wf** [comment]
+1

**Comment by trentgrover-wf** [comment]
+1
@jayudey-wf ready for merge (docs only changes)

**Comment by maxwellpeterson-wf** [comment]
+1

**Comment by evanweible-wf** [comment]
@trentgrover-wf @maxwellpeterson-wf @dustinlessard-wf squashed some commits.

**Comment by dustinlessard-wf** [comment]
+1

**Comment by maxwellpeterson-wf** [comment]
+1

**Comment by evanweible-wf** [comment]
@jayudey-wf ready for merge

**Comment by jayudey-wf** [comment]
QA +1, doc change only

Merging into master

**Pull request merged**